pyccel/ast/class_defs.py

```diff
@@ -137,7 +136,7 @@
 
 #=======================================================================================
 
-ListClass = ClassDef('list', class_type = NativeHomogeneousList(),
+ListClass = ClassDef('list',
         methods=[
             PyccelFunctionDef('append', func_class = ListAppend,
                 decorators = {}),
```

Comment:
I'm confused about why you decided to remove `class_type`.

Reply:
In the new implementation the class type contains all information about the type. So it is no longer possible to initialise `NativeHomogeneousList()`, instead we need to do something like `NativeHomogeneousList(PythonNativeInt())` or `NativeHomogeneousList(NumpyNDArray(PythonNativeFloat()))`. As this class is valid for all the different kinds of lists that you can imagine I thought it was better to not provide this argument rather than creating a duplicate of this class on the fly for each type that is instantiated.

Reply:
What about `StringType`?

Reply:
For `StringType` there is only one type. In Python it is not possible to have a string of chars vs string of unsigned chars. There are just strings so the class `StringType` doesn't take an argument. This means that it is possible to provide the `class_type` argument